cmd/backup/main.go

```diff
@@ -4,19 +4,28 @@
 package main
 
 import (
+	"flag"
 	"fmt"
+	"log"
 	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/robfig/cron/v3"
 )
 
-func main() {
-	s, err := newScript()
+func runBackup(c *Config) {
+	s, err := newScript(c)
 	if err != nil {
 		panic(err)
 	}
 
 	unlock, err := s.lock("/var/lock/dockervolumebackup.lock")
 	defer s.must(unlock())
-	s.must(err)
+	if err != nil {
+		log.Println("Error during locking:", err)
```

Comment:
This raises an interesting question: so far a backup run has an exclusive lock mostly because I was too lazy to deal with the consequences of concurrent invocations. Do you think we should keep it like this or remove this limitation?

Reply:
I would keep it for now, I do not see a solution to create a more fine grade solution. Worst case is that a job gets blocked by another for some time.

It could be moved into seperate issue, but I am not totally sure how locks could be more finer graded. A container could use multiple volumes each with their own schedule. It would be possible that both want to stop/start at the same time...